src/net_processing.cpp

```diff
@@ -816,6 +816,23 @@ void Misbehaving(NodeId pnode, int howmuch, const std::string& message) EXCLUSIV
         LogPrint(BCLog::NET, "%s: %s peer=%d (%d -> %d)%s\n", __func__, state->name, pnode, state->nMisbehavior-howmuch, state->nMisbehavior, message_prefixed);
 }
 
+static bool MayResultInDisconnect(const CValidationState& state, bool via_compact_block) {
+    return (state.GetDoS() > 0);
+}
+
+static bool MaybePunishNode(NodeId nodeid, const CValidationState& state, bool via_compact_block, const std::string& message = "") {
+    int nDoS = state.GetDoS();
+    if (nDoS > 0 && !via_compact_block) {
+         LOCK(cs_main);
```

Comment:
In commit "[refactor] Use maybepunish etc" (8226bed4191a50129ac6fdbcb8fad5e1c6b7cacd)

Note: This acquires lock recursively in `PeerLogicValidation::BlockChecked`. Seems fine, but just wanted to note it wasn't happening before.